src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -626,6 +636,161 @@ class MessageHelperTest : public ::testing::Test {
+    MockApplicationManagerSettings mock_app_mngr_settings;
+    EXPECT_CALL(mock_application_manager_, GetNextHMICorrelationID())
+        .WillOnce(Return(kCorrelationId));
+    EXPECT_CALL(mock_application_manager_, get_settings())
+        .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+    EXPECT_CALL(mock_app_mngr_settings, video_server_type())
+        .WillRepeatedly(ReturnRef(video_server_type));
+    if (video_server_type == "socket") {
+      EXPECT_CALL(mock_app_mngr_settings, server_address())
+          .WillOnce(ReturnRef(server_path));
+      EXPECT_CALL(mock_app_mngr_settings, video_streaming_port())
+          .WillOnce(Return(port));
+    } else if (video_server_type == "pipe") {
+      EXPECT_CALL(mock_app_mngr_settings, named_video_pipe_path())
+          .WillOnce(ReturnRef(server_path));
+    } else {
+      EXPECT_CALL(mock_app_mngr_settings, video_stream_file())
+          .WillOnce(ReturnRef(server_path));
+    }
+    smart_objects::SmartObjectSPtr result;
+    EXPECT_CALL(mock_application_manager_, ManageHMICommand(_))
+        .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+    MessageHelper::SendNaviStartStream(kAppId, mock_application_manager_);
+
+    ASSERT_TRUE(result.valid());
+    EXPECT_TRUE((*result)[strings::params].keyExists(strings::function_id));
+    EXPECT_EQ((*result)[strings::params][strings::function_id].asInt(),
+              hmi_apis::FunctionID::Navigation_StartStream);
+    CheckParamsOfMessage(result, true, true);
+    EXPECT_TRUE((*result)[strings::msg_params].keyExists(strings::url));
+    EXPECT_TRUE((*result)[strings::msg_params][strings::url].asString() == url);
+  }
+
+  /**
+   * @brief Check common parameters from messag
+   * @param result contains message which need to check
+   * @param is_app_id flag for checking application id parameter
+   * @param is_corr_id flag for checking correlation id parameter
+   */
+  void CheckParamsOfMessage(smart_objects::SmartObjectSPtr result,
+                            bool is_app_id,
+                            bool is_corr_id) {
+    if (is_app_id) {
+      EXPECT_TRUE((*result)[strings::msg_params].keyExists(strings::app_id));
+      EXPECT_EQ((*result)[strings::msg_params][strings::app_id].asInt(),
+                kAppId);
+    }
+
+    if (is_corr_id) {
+      EXPECT_TRUE(
+          (*result)[strings::params].keyExists(strings::correlation_id));
+      EXPECT_EQ(kCorrelationId,
+                (*result)[strings::params][strings::correlation_id].asInt());
+    }
+  }
+
+  /**
+   * @brief Fill structure for testing
+   * @param obj structure wich need to fill
+   * @param group_alias parameter from structure
+   * @param group_name parameter from structure
+   * @param group_id parameter from structure
+   * @param state parameter from structure
+   */
+  void FillFunctionalGroupPermissionObj(policy::FunctionalGroupPermission& obj,
+                                        const std::string group_alias,
```

Comment:
@dev-gh I do not see the point because I send const char* for wich compiler creates temporary objects